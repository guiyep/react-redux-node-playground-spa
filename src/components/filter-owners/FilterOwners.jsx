import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Input } from 'antd';

const FilterOwners = memo(({ onChange, value }) => {
  return <Input placeholder="Filter owners..." defaultValue={value} onChange={onChange} />;
});

FilterOwners.displayName = 'FilterOwners';

FilterOwners.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
};

FilterOwners.defaultProps = {
  value: undefined,
};

export default FilterOwners;
