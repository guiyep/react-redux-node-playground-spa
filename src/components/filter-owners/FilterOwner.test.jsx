import React from 'react';
import { create } from 'react-test-renderer';
import FilterOwners from './FilterOwners';
import ReactDOM from 'react-dom';

describe('FilterOwners', () => {
  test('it matches the snapshot', () => {
    const component = create(<FilterOwners onChange={() => {}} value="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FilterOwners onChange={() => {}} value="test" />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
