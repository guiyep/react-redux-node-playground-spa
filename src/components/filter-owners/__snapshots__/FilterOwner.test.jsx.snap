// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterOwners it matches the snapshot 1`] = `
<input
  className="ant-input"
  disabled={false}
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="Filter owners..."
  type="text"
  value="test"
/>
`;
